airbyte-integrations/connectors/destination-google-sheets/destination_google_sheets/helpers.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+import re
+from typing import List
+
+from pygsheets import Spreadsheet, Worksheet
+from pygsheets.exceptions import WorksheetNotFound
+
+
+def get_spreadsheet_id(id_or_url: str) -> str:
+    if re.match(r"(http://)|(https://)", id_or_url):
+        m = re.search(r"(/)([-\w]{40,})([/]?)", id_or_url)
+        if m.group(2):
+            return m.group(2)
+    else:
+        return id_or_url
+
+
+class ConnectionTest:
+
+    """
+    Performs connection test write operation to ensure the target spreadsheet is available for writing.
+    Initiating the class itself, performs the connection test and stores the result in ConnectionTest.result property.
+    """
+
+    def __init__(self, spreadsheet: Spreadsheet):
+        self.spreadsheet = spreadsheet
+        self.wks_name: str = "_airbyte_conn_test"
+        self.test_data: List[str] = ["conn_test", "success"]
+        self.result = self.perform_connection_test()
```

Comment:
nit: creating an instance of this shouldn't really imply performing the connection test imo. We should call that manually.

Reply:
Done.